### PR TITLE
Contact list UI 

### PR DESCRIPTION
**Pull request by TdVBd**
updated Private Chats UI
updated DirectMessageScreen UI and functions
added a Contact Settings screen, with deletion and edition functions
updated mockdatabase to incorporate contactlist
updated contact list implementation to have IDs match the private chat IDs
reformated some SharedUI and ChatScreen components and corrected their usages
some other minor fixes

### CODE REVIEW

File: app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt [hunk 1]
```diff
@@ -981,33 +974,42 @@ class DatabaseConnection : DbRepository {
...
         val newChatId = UUID.randomUUID().toString()
+        contactID = newChatId
         val memberPath = getPrivateChatMembersPath(newChatId)
         val members = mapOf(currentUserUID to true, otherUID to true)
         rtDb
             .getReference(memberPath)
             .updateChildren(members)
             .addOnSuccessListener {
               Log.d("DatabaseConnect", "startDirectMessage : Members successfully added!")
+              contactsViewModel.createContact(otherUID, contactID)
```
Comment (cemuelle) [inline app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt:999]: A viewModel has nothing to do in the DatabaseConnection. 
You can call the DatabaseConnection from the viewModel, but not the other direction

Comment (TdVBd) [reply app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt:999]: this is a temporary fix to a syncronization problem, thanks for pointing it out i could have forgot it

Comment (TdVBd) [reply app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt:1047]: i found a way to solve this while implementing the contact request, so you will see it solved in the next PR

**Comment by TdVBd** [comment]
Lots of files updated due to a merging error 
Lots of files with minor updates due to refactoring and reformating

File: app/src/androidTest/java/com/github/se/studybuddies/tests/ChatTest.kt
Comment (hanelmouta) [inline app/src/androidTest/java/com/github/se/studybuddies/tests/ChatTest.kt:1]: why did you comment those test ? do they no longer pass?

File: app/src/androidTest/java/com/github/se/studybuddies/tests/LoginTest.kt
Comment (hanelmouta) [inline app/src/androidTest/java/com/github/se/studybuddies/tests/LoginTest.kt:1]: same for here

File: app/src/main/java/com/github/se/studybuddies/MainActivity.kt [hunk 2]
```diff
@@ -412,17 +421,20 @@ class MainActivity : ComponentActivity() {
...
-                    backStackEntry ->
-                  val topicFileID = backStackEntry.arguments?.getString("topicFileID")
-                  if (topicFileID != null) {
-                    val topicFileViewModel = remember { TopicFileViewModel(topicFileID, db) }
-                    TopicResources(topicFileID, topicFileViewModel, navigationActions)
-                    Log.d("MyPrint", "Successfully navigated to TopicResources")
-                  }
-                }
+
+            /*composable(
```
Comment (hanelmouta) [inline app/src/main/java/com/github/se/studybuddies/MainActivity.kt:425]: you can delete this if not used

File: app/src/main/java/com/github/se/studybuddies/navigation/Destinations.kt [hunk 3]
```diff
@@ -13,7 +13,6 @@ val GROUPS_SETTINGS_DESTINATIONS =
     listOf(
         Destination(route = Route.GROUPSETTING, textId = "Modify group"),
         Destination(route = Route.GROUPMEMBERS, textId = "Members"),
-        Destination(route = Route.GROUPMEMBERADD, textId = "Add member"),
```
Comment (hanelmouta) [inline app/src/main/java/com/github/se/studybuddies/navigation/Destinations.kt:16]: why did this got deleted ?

File: app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt
Comment (hanelmouta) [inline app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt:1]: the call button still leads to a placeholder screen , this should not take too long to implement.

File: app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                    modifier =
+                        Modifier.clip(RoundedCornerShape(4.dp))
+                            .width(100.dp)
+                            .height(40.dp)
+                            .testTag(contactID + "_delete_no_button"),
+                    colors =
+                        ButtonDefaults.buttonColors(containerColor = Blue, contentColor = White)) {
+                      Text(
+                          text = stringResource(R.string.cancel),
+                          modifier = Modifier.testTag(contactID + "_delete_no_text"))
```
Comment (hanelmouta) [inline app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt:242]: use string

File: app/src/main/java/com/github/se/studybuddies/viewModels/ContactsViewModel.kt
Comment (hanelmouta) [inline app/src/main/java/com/github/se/studybuddies/viewModels/ContactsViewModel.kt:1]: you should clear the log some of them are not clear you should clean them

File: app/src/main/java/com/github/se/studybuddies/viewModels/GroupViewModel.kt [hunk 5]
```diff
@@ -97,29 +81,8 @@ class GroupViewModel(
     viewModelScope.launch { db.deleteGroup(groupUID) }
   }
 
-  fun addUserToGroup(groupUID: String, text: String = "", callBack: (Boolean) -> Unit) {
```
Comment (hanelmouta) [inline app/src/main/java/com/github/se/studybuddies/viewModels/GroupViewModel.kt:100]: why did you delete this

**Review by hanelmouta — CHANGES_REQUESTED** [review]
I reviewed you pr you did a nice job however I think when you merge you had some issue

File: app/src/main/java/com/github/se/studybuddies/data/TopicItem.kt
Comment (TdVBd) [inline app/src/main/java/com/github/se/studybuddies/data/TopicItem.kt:1]: conflict resolution caused a back and forth, no new code

File: app/src/main/java/com/github/se/studybuddies/ui/groups/GroupMembers.kt
Comment (TdVBd) [inline app/src/main/java/com/github/se/studybuddies/ui/groups/GroupMembers.kt:1]: unsure of how to resolt conflict, i commented the code

File: app/src/main/java/com/github/se/studybuddies/viewModels/ContactsViewModel.kt
Comment (TdVBd) [reply app/src/main/java/com/github/se/studybuddies/viewModels/ContactsViewModel.kt:1]: will do in future PR about code cleaning

File: app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt (hunk 4 above)
Comment (TdVBd) [reply app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt:242]: might have time to reformat this next PR

File: app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt
Comment (TdVBd) [reply app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt:1]: I have to check with lorie how the video call is implemented and i'll add this next PR

File: app/src/androidTest/java/com/github/se/studybuddies/tests/ChatTest.kt
Comment (TdVBd) [reply app/src/androidTest/java/com/github/se/studybuddies/tests/ChatTest.kt:1]: yes since i changed the chat structure quite a bit

**Comment by cemuelle** [comment]
I just test you're branch on the chat and it's appear that the button to be able to search in the message and to filter by type of message.
It seems to have been replace by a button that lead to a placeholder screen. And on this screen the goback button leads to the solo study home and not back as it should

**Comment by TdVBd** [comment]
@cemuelle goback button are all incorrectly implemented for now as they lead to specified route instead of calling nav.goBack(), i deal with this issue globally in my reformating PR
for the filtering, i have adapted for it to fit the figma, but i see i removed it from the groups chat too, this is a quick fix

File: app/src/main/java/com/github/se/studybuddies/ui/groups/GroupSetting.kt
Comment (TdVBd) [inline app/src/main/java/com/github/se/studybuddies/ui/groups/GroupSetting.kt:1]: does not correspond to what is on my branch after push for some reason

**Review by hanelmouta — APPROVED** [review]
